processing/src/main/java/io/druid/query/groupby/having/DimensionSelectorHavingSpec.java

```diff
@@ -81,12 +97,28 @@ public boolean eval(Row row)
   {
     byte[] dimBytes = StringUtils.toUtf8(this.dimension);
     byte[] valBytes = StringUtils.toUtf8(this.value);
-    return ByteBuffer.allocate(2 + dimBytes.length + valBytes.length)
-                     .put(CACHE_KEY)
-                     .put(dimBytes)
-                     .put(STRING_SEPARATOR)
-                     .put(valBytes)
-                     .array();
+    byte[] extractionFnBytes = null;
+
+    if (this.getExtractionFn() != null) {
+      extractionFnBytes = this.getExtractionFn().getCacheKey();
+    }
+
+    if (extractionFnBytes != null) {
```

Comment:
you can set extractionFnBytes to an empty array instead of null and simplify the if/else loop